integration/flink/src/main/java/io/openlineage/flink/OpenLineageFlinkJobListener.java

```diff
@@ -66,23 +89,26 @@ private void makeTransformationsArchivedList(StreamExecutionEnvironment executio
       ArrayList<Transformation<?>> previousTransformationList =
           (ArrayList<Transformation<?>>)
               FieldUtils.readField(transformations, executionEnvironment, true);
-      List<Transformation<?>> transformationList = new ArchivedList<>(previousTransformationList);
+      List<Transformation<?>> transformationList =
+          new UnclearableList<>(
+              Optional.ofNullable(previousTransformationList).orElse(new ArrayList<>()));
       FieldUtils.writeField(transformations, executionEnvironment, transformationList, true);
     } catch (IllegalAccessException e) {
       log.error("Failed to rewrite transformations");
     }
   }
 
-  static class ArchivedList<T> extends ArrayList<T> {
-    @Getter List<T> archive;
+  static class UnclearableList<T> extends ArrayList<T> {
```

Comment:
Why? It's clearable, just stores it's value in additional variable. 

BTW, I changed it to `ArchivedList` after your suggestion 😉